### PR TITLE
SONAR-7330 Optimize Active Rule Load & Drop RuleNormalizer

### PR DESCRIPTION
**Pull request by teryk**

### CODE REVIEW

File: sonar-db/src/main/java/org/sonar/db/rule/RuleDtoFunctions.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  public static Function<RuleDto, Integer> toId() {
+    return ToId.INSTANCE;
+  }
+
+  private enum ToId implements Function<RuleDto, Integer> {
+    INSTANCE;
+
+    @Override
+    public Integer apply(@Nonnull RuleDto rule) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/rule/RuleDtoFunctions.java]: MAJOR The "rule" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

File: sonar-db/src/main/java/org/sonar/db/qualityprofile/ActiveRuleDao.java [hunk 2]
```diff
@@ -206,4 +210,17 @@ private ParamIdToDto(ActiveRuleMapper mapper) {
...
+
+  private static class RuleIdToDto implements Function<List<Integer>, List<ActiveRuleDto>> {
+    private final ActiveRuleMapper mapper;
+
+    private RuleIdToDto(ActiveRuleMapper mapper) {
+      this.mapper = mapper;
+    }
+
+    @Override
+    public List<ActiveRuleDto> apply(@Nonnull List<Integer> partitionOfRuleIds) {
```
Comment (SonarTech) [inline sonar-db/src/main/java/org/sonar/db/qualityprofile/ActiveRuleDao.java]: MAJOR The "partitionOfRuleIds" parameter nullability is different in the superclass method, and that should not be changed. [rule](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2638)

**Comment by SonarTech** [comment]
SonarQube analysis reported 2 issues:
* MAJOR 2 major

Watch the comments in this conversation to review them.

**Comment by julienlancelot** [comment]
Some feedback to apply, then LGTM

**Pull request closed without merging**